File: src/main/java/red/tetracube/smarthub/configurations/properties/SmartHubConfig.java
package red.tetracube.smarthub.configurations.properties;

import io.quarkus.arc.config.ConfigProperties;
import org.eclipse.microprofile.config.inject.ConfigProperty;

@ConfigProperties(prefix = "smart-hub")
public interface SmartHubConfig {

    @ConfigProperty(name = "iot")
    IoTConfig iot();
}

File: src/main/kotlin/red/tetracube/smarthub/configurations/properties/SmartHubConfig.kt
package red.tetracube.smarthub.configurations.properties

import io.quarkus.arc.config.ConfigProperties
import org.eclipse.microprofile.config.inject.ConfigProperty

@ConfigProperties(prefix = "smart-hub")
interface SmartHubConfig{

    @ConfigProperty(name = "iot")
    fun iot(): IoTConfig
}
